src/modules/create_institution/app/create_institution_usecase.ts

```diff
@@ -0,0 +1,38 @@
+import { randomUUID } from 'crypto';
+import { Institution } from "../../../core/structure/entities/Institution";
+import { MissingParameter } from '../../../core/helpers/errors/ModuleError';
+import { IInstitutionRepo } from '../../../core/repositories/interfaces/IInstitutionRepo';
+
+export class CreateInstitutionUsecase {
+    public database_repo: IInstitutionRepo;
+
+    constructor(database_repo: IInstitutionRepo) {
+        this.database_repo = database_repo;
+    }
+
+    public async execute(institution: Institution){
```

Comment:
Here u check if the headers has the authorization, check if the user is authenticated, and them u build the Institution class and create!